webapp/components/features/FollowList/FollowList.story.js

```diff
@@ -0,0 +1,111 @@
+import { storiesOf } from '@storybook/vue'
+import { withA11y } from '@storybook/addon-a11y'
+import { action } from '@storybook/addon-actions'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+import fuzzyFilterUser from './FollowList.story.json'
+
+helpers.init()
+
+const user = {
+  name: 'Jenny Rostock',
+  id: 'u3',
+  followedByCount: 12,
+  followedBy: helpers.fakeUser(7),
+  followingCount: 28,
+  following: helpers.fakeUser(7),
+}
+
+const lessThanSevenUser = {
+  ...user,
+  followedByCount: 3,
+  followedBy: user.followedBy.slice(0, 3),
+  followingCount: 3,
+  following: user.following.slice(0, 3),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [...user.followedBy, ...helpers.fakeUser(5)],
+  following: [...user.following, ...helpers.fakeUser(21)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedBy: [],
+  followedByCount: 0,
+  following: [],
+  followingCount: 0,
+}
+
+storiesOf('FollowList', module)
+  .addDecorator(withA11y)
+  .addDecorator(helpers.layout)
+  .add('without connections', () => {
+    return {
+      components: { FollowList },
+      store: helpers.store,
+      data() {
+        return { user: noConnectionsUser }
+      },
+      template: '<follow-list :user="user" type="following" />',
+    }
+  })
+  .add('with all connections loaded', () => {
+    return {
+      components: { FollowList },
+      store: helpers.store,
+      data() {
+        return { user: lessThanSevenUser }
+      },
+
+      template: '<follow-list :user="user"/>',
+    }
+  })
+
+  .add('with more connections loadable', () => {
+    return {
+      components: { FollowList },
+      store: helpers.store,
+      data() {
+        return { user: { ...user } }
+      },
+      methods: {
+        fetchAllConnections(type) {
+          this.user[type] = allConnectionsUser[type]
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-04-07T14:36:19Z" title="Tuesday, April 7th 2020, 4:36:19 pm +02:00">Apr 7, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,123 @@
+import { storiesOf } from '@storybook/vue'
+import { withA11y } from '@storybook/addon-a11y'
+import { action } from '@storybook/addon-actions'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+import fuzzyFilterUser from './FollowList.story.json'
+
+helpers.init()
+
+const user = {
+  name: 'Jenny Rostock',
+  id: 'u3',
+  followedByCount: 12,
+  followedBy: helpers.fakeUser(7),
+  followingCount: 28,
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [...user.followedBy, ...helpers.fakeUser(5)],
+  following: [...user.following, ...helpers.fakeUser(21)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedBy: [],
+  followedByCount: 0,
+  following: [],
+  followingCount: 0,
+}
+
+const wrapTemplates = (templates) => `
+<div style="display: flex; flex-wrap: wrap;">
+${templates.map((template) => `<div style="margin: 8px;">${template}</div>`).join('')}
+</div>
+`
+
+storiesOf('FollowList', module)
+  .addDecorator(withA11y)
+  .addDecorator(helpers.layout)
+  .add('without connections', () => {
+    return {
+      components: { FollowList },
+      store: helpers.store,
+      data() {
+        return { user: noConnectionsUser }
+      },
+      template: wrapTemplates([
+        '<follow-list :user="user" />',
+        '<follow-list :user="user" type="followedBy" />',
+      ]),
+    }
+  })
+  .add('with up to 7 connections', () => {
+    return {
+      components: { FollowList },
+      store: helpers.store,
+      data() {
+        return { user: { ...user } }
+      },
+      methods: {
+        fetchAllConnections(type) {
+          this.user[type] = allConnectionsUser[type]
+          action('fetchAllConnections')(type, this.user)
+        },
+      },
+      template: wrapTemplates([
+        '<follow-list :user="user" @fetchAllConnections="fetchAllConnections"/>',
+        '<follow-list :user="user" type="followedBy" @fetchAllConnections="fetchAllConnections"/>',
+      ]),
+    }
+  })
+
+  .add('with all connections', () => {
```


I'm a little confused by these stories.

`with up to 7 connections` has more than 7 connections, they're just not expanded yet. `with all connections` is the same thing expanded – but I can achieve the same thing by clicking the buttons in the first story, so it's kind of redundant.

Instead, it would be cool to have a list that has _actually_ less than 7 connections, so we see that it doesn't display the button in this case. The above two cases can be covered with just one story, which could be called `with more than 7 connections` or `default`, even.

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-09T20:56:34Z" title="Thursday, April 9th 2020, 10:56:34 pm +02:00">Apr 9, 2020</time>_

---


The profile page had the count of displayed connections hardcoded to 7; I just took that over, when I started.
It makes way more sense now to introduce a `hasMore` computed value of `user.followingCount > user.following.length`

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-09T21:56:11Z" title="Thursday, April 9th 2020, 11:56:11 pm +02:00">Apr 9, 2020</time>_

---


e018e2ead
30f66c3e1